Domain/Dto/Bets.cs

```diff
@@ -10,12 +10,17 @@ public class Bets
         [StringLength(100)]
         public string Name { get; set; }
 
-        public DateTime Date { get; private set; } = TimeProvider.System.GetUtcNow().DateTime;
+        public DateTime Date { get; private set; }
 
         [Required]
         public Guid BetableEntityA { get; set; }
 
         [Required]
         public Guid BetableEntityB { get; set; }
+
+        public void SetTime(TimeProvider timeProvider)
+        {
+            Date = timeProvider.GetUtcNow().DateTime;
+        }
```

Comment:
let's remove this - we can set the property from the service

Reply:
It's private the constructor